package/utils/uencrypt/src/uencrypt-mbedtls.c

```diff
@@ -0,0 +1,212 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later
+ * Copyright (C) 2023 Eneas Ulir de Queiroz
+ */
+
+#include <ctype.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <mbedtls/cipher.h>
+
+int do_crypt(FILE *infile, FILE *outfile, const mbedtls_cipher_info_t *cipher_info,
+	     const unsigned char *key, const unsigned char *iv, int enc, int padding)
+{
+    mbedtls_cipher_context_t ctx;
+    unsigned char inbuf[1024], outbuf[1024 + MBEDTLS_MAX_BLOCK_LENGTH];
+    size_t inlen, outlen;
+    int ret = 0;
+    int step;
+
+    mbedtls_cipher_init(&ctx);
+    if ((ret = mbedtls_cipher_setup(&ctx, cipher_info))) {
+	fprintf(stderr, "Error: mbedtls_cipher_setup: %d\n", ret);
+	goto out;
+    }
+    step = iv ? 1024 : mbedtls_cipher_get_block_size(&ctx);
+    if ((ret = mbedtls_cipher_setkey(&ctx, key, (int) mbedtls_cipher_get_key_bitlen(&ctx),
+			      enc ? MBEDTLS_ENCRYPT : MBEDTLS_DECRYPT))) {
+	fprintf(stderr, "Error: mbedtls_cipher_setkey: %d\n", ret);
+	goto out;
+    }
+    if (iv && (ret = mbedtls_cipher_set_iv(&ctx, iv, 16))) {
```

Comment:
This assumes an ívlen of 16. I think  you should use:
````
mbedtls_cipher_set_iv(&ctx, iv, cipher_info->iv_size)
````

Reply:
Done.